lib/esbonio/esbonio/lsp/__init__.py

```diff
@@ -179,11 +182,23 @@ def on_code_action(ls: RstLanguageServer, params: CodeActionParams):
 
         return actions
 
+    @server.feature(HOVER)
+    def on_hover(ls: RstLanguageServer, params: HoverParams):
+        uri = params.text_document.uri
+        doc = ls.workspace.get_document(uri)
+        pos = params.position
+        context = HoverContext(doc=doc, position=pos)
+
+        for feature in ls._features.values():
+            hovers = feature.hover(context)
```

Comment:
There should be a check if the feature has a hover function.

Or are feature functions like this always there, because they are defined with an empty body, which got inherited from a common FEATURE-class?

Reply:
you need to add the hovers from each hover to a common list and return them together.
Right now only the hovers from the latest features get returned